From bbf535383e3e04dbfda2b3104c4ee11689140b54 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 4 Nov 2019 15:30:55 +0100
Subject: [PATCH] qeglfsintegration: stop using deprecated and in 5.13 removed
 screen API
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

* first they were deprecated in 5.12.3 with:
  commit 01e1df90a7debd333314720fdd5cf6cd9964d796
  Author: Tor Arne Vestbø <tor.arne.vestbo@qt.io>
  Date:   Thu Mar 14 00:36:34 2019 +0100

    Move screen maintenance functions from QPlatformIntegration to QWSI

    QWindowSystemInterface is the de facto API for any plumbing going from
    the platform plugin to QtGui. Having the functions as protected members
    of QPlatformIntegration was idiosyncratic, and resulted in awkward
    workarounds to be able to call the functions from outside of the
    QPlatformIntegration subclass.

    The functions in QPlatformIntegration have been left in, but deprecated
    so that platform plugins outside of qtbase have a chance to move over to
    the new QWSI API before they are removed.

    Change-Id: I327fec460db6b0faaf0ae2a151c20aa30dbe7182
    Reviewed-by: Gatis Paeglis <gatis.paeglis@qt.io>

* then completely removed in 5.13.0 with:
  commit 7a64ffb7738dc975b5008800901c8cd8ab238a0f
  Author: Tor Arne Vestbø <tor.arne.vestbo@qt.io>
  Date:   Thu Mar 21 12:01:51 2019 +0100

    Remove deprecated screen maintenance functions in QPlatformIntegration

    The logic for removing QScreens from QGuiApplicationPrivate has been
    moved into the QScreen destructor, similar to QWindow.

    Change-Id: I18ad57d8dcf9f765c47be7c082bf075af3ebe69c
    Reviewed-by: Friedemann Kleint <Friedemann.Kleint@qt.io>
    Reviewed-by: Lars Knoll <lars.knoll@qt.io>
---
 hwcomposer/qeglfsintegration.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hwcomposer/qeglfsintegration.cpp b/hwcomposer/qeglfsintegration.cpp
index 112bfd8..4262f1d 100644
--- a/hwcomposer/qeglfsintegration.cpp
+++ b/hwcomposer/qeglfsintegration.cpp
@@ -107,14 +107,20 @@ QEglFSIntegration::QEglFSIntegration()
     }
 
     mScreen = new QEglFSScreen(mHwc, mDisplay);
+#if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)
     screenAdded(mScreen);
+#else
+    QWindowSystemInterface::handleScreenAdded(mScreen);
+#endif
 
     mInputContext = QPlatformInputContextFactory::create();
 }
 
 QEglFSIntegration::~QEglFSIntegration()
 {
-#if QT_VERSION >= 0x050500
+#if QT_VERSION >= QT_VERSION_CHECK(5, 13, 0)
+    QWindowSystemInterface::handleScreenRemoved(mScreen);
+#elif QT_VERSION >= QT_VERSION_CHECK(5, 5, 0)
     destroyScreen(mScreen);
 #else
     delete mScreen;
